implementation/src/main/java/io/smallrye/mutiny/converters/uni/UniToMultiPublisher.java

```diff
@@ -0,0 +1,94 @@
+package io.smallrye.mutiny.converters.uni;
+
+import static io.smallrye.mutiny.helpers.EmptyUniSubscription.CANCELLED;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.subscription.UniSubscriber;
+import io.smallrye.mutiny.subscription.UniSubscription;
+
+public class UniToMultiPublisher<T> implements Publisher<T> {
+
+    private final Uni<T> uni;
+
+    public UniToMultiPublisher(Uni<T> uni) {
+        this.uni = uni;
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super T> downstream) {
+        downstream.onSubscribe(new UniToMultiSubscription<>(uni, downstream));
+    }
+
+    private static class UniToMultiSubscription<T> implements UniSubscription, Subscription, UniSubscriber<T> {
+
+        private final Uni<T> uni;
+        private final Subscriber<? super T> downstream;
+        private final AtomicReference<UniSubscription> upstream = new AtomicReference<>();
+
+        private UniToMultiSubscription(Uni<T> uni, Subscriber<? super T> downstream) {
+            this.uni = uni;
+            this.downstream = downstream;
+        }
+
+        @Override
+        public void cancel() {
+            UniSubscription sub;
+            synchronized (upstream) {
+                sub = upstream.getAndSet(CANCELLED);
+            }
+            if (sub != null) {
+                sub.cancel();
+            }
+        }
+
+        @Override
+        public void request(long n) {
+            synchronized (upstream) {
+                if (n <= 0L) {
+                    downstream.onError(new IllegalArgumentException("Invalid request"));
+                    return;
+                }
+                if (upstream.get() == CANCELLED) {
+                    return;
+                }
+                uni.subscribe().withSubscriber(this);
+            }
+        }
+
+        @Override
+        public void onSubscribe(UniSubscription subscription) {
+            if (!upstream.compareAndSet(null, subscription)) {
+                downstream.onError(new IllegalStateException(
+                        "Invalid subscription state - already have a subscription for upstream"));
+            }
+        }
+
+        @Override
+        public void onItem(T item) {
+            if (upstream.getAndSet(CANCELLED) != CANCELLED) {
+                if (item != null) {
+                    try {
+                        downstream.onNext(item);
+                    } catch (Throwable failure) {
+                        downstream.onError(failure);
```

Comment:
I don't believe we should call onError here, if the failure comes from the downstream. You need to cancel, but I don't think we need to send a second event.

Reply:
If we just do:

```java
if (upstream.getAndSet(CANCELLED) != CANCELLED) {
    if (item != null) {
      downstream.onNext(item);
    }
    downstream.onComplete();
}
```

then tests fail in `UniRepeatTest` and `UniCastProcessorTest` fail.

Handling exceptions as dropped exceptions does not work either, the downstream in these tests does want the error:

```java
if (upstream.getAndSet(CANCELLED) != CANCELLED) {
    if (item != null) {
        try {
            downstream.onNext(item);
        } catch (Throwable failure) {
            Infrastructure.handleDroppedException(failure);
            return;
        }
    }
    downstream.onComplete();
}
```

Reply:
(basically calling `onError` does preserve the existing behaviour)